src/middlewared/middlewared/plugins/nfs.py

```diff
@@ -108,6 +108,28 @@ def nfs_compress(self, nfs):
                         "domain"
                     )
 
+        if new["v4"] and new["v4_krb"] and await self.middleware.call('activedirectory.get_state') != "DISABLED":
+            """
+            In environments with kerberized NFSv4 enabled, we need to tell winbindd to not prefix
+            usernames with the short form of the AD domain. Directly update the db and regenerate
+            the smb.conf to avoid having a service disruption due to restarting the samba server.
+            """
+            ad = await self.middleware.call('activedirectory.config')
+            await self.middleware.call(
+                'datastore.update',
+                'directoryservice.activedirectory',
+                ad['id'],
+                {'ad_use_default_domain': True}
+            )
+            await self.middleware.call('etc.generate', 'smb')
+            await self.middleware.call('service.reload', 'cifs') 
+            if await self.middleware.call('smb.get_smb_ha_mode') == 'LEGACY':
+                verrors.add(
```

Comment:
We should probably raise a validation error immediately before reconfiguring the AD service and reloading samba.

Reply:
We should wrap the call to `smb.get_smb_ha_mode` in an `if else` statement and raise a validation error when smb is configured in LEGACY mode.